mobile/bazel/envoy_mobile_repositories.bzl

```diff
@@ -97,7 +97,7 @@ def kotlin_repos():
 def android_repos():
     http_archive(
         name = "build_bazel_rules_android",
-        urls = ["https://github.com/bazelbuild/rules_android/archive/refs/tags/v0.1.1.zip"],
-        sha256 = "cd06d15dd8bb59926e4d65f9003bfc20f9da4b2519985c27e190cddc8b7a7806",
-        strip_prefix = "rules_android-0.1.1",
+        urls = ["https://github.com/bazelbuild/rules_android/archive/refs/tags/v0.6.0.zip"],
+        sha256 = "691c90615ebf66aa474fdd5e17b92d54f62e5e3627f4eb224e9321cfcf5fe83f",
+        strip_prefix = "rules_android-0.6.0",
```

Comment:
Seeing more `/refs/tags/` creeping into the dependency URLs which makes sens as that is what GitHub uses on the page now. The links work without it though and are cleaner. Almost all the links omit the `/refs/tags/`. Wondering if we should standardardize?

As per dependency standards we should use the maintainer provided binary here

```
        urls = ["https://github.com/bazelbuild/rules_android/releases/download/v0.6.0/rules_android-v0.6.0.tar.gz"],
        sha256 = "af84b69ab3d16dd1a41056286e6511f147a94ccea995603e13e934c915c1631c",
        strip_prefix = "rules_android-0.6.0",
```

Reply:
yeah - that works in this case so happy to update

in the general case refs/tags is preferable to direct archive downloads - although the latter is the only option where the upstream doesnt maintain regular releases and/or we want/need to pin to a particular commit

non-releasing upstream are a big pita - using hashes is very insecure because you have to actually check the commit is a real commit on the branch you are expecting - you can use those addresses with _any_ commit in a fork branch without creating any pr or ever landing any commits on the target repos branches